proxy/src/main.rs

```diff
@@ -26,9 +26,39 @@ mod server_warp;
 /// Origin of data required like the on-chain Registry.
 mod source;
 
+/// Convenience function for setting env vars only if they are unset
+fn set_if_unset (key: &str, val: &str) {
+    match std::env::var(key) {
+        Ok(_val) => { /* nothing to do */ },
+        Err(_e) => { std::env::set_var(key, val); },
+    }
+}
+
+#[test]
+fn test_set_if_unset() {
+    match std::env::var("DUMMY_VALUE") {
+        Ok(_val) => {
+            // should be unset
+            panic!("$DUMMY_VALUE should be unset.")
+        },
+        Err(_e) => {
+            set_if_unset("DUMMY_VALUE", "hello world")
+        }
+    }
+    match std::env::var("DUMMY_VALUE") {
+        Ok(val) => { assert_eq!(val, "hello world") },
+        Err(_e) => {
+            panic!("$DUMMY_VALUE is still unset.")
+        }
+    }
+}
+
 fn main() {
-    std::env::set_var("RUST_BACKTRACE", "full");
-    std::env::set_var("RUST_LOG", "info");
+    // enable backtrace if it isn't already
```

Comment:
Think we can omit all these comment as the line they are referring to are self-explanatory.